Dockerfile

```diff
@@ -1,145 +1,166 @@
 #--------------------------------------
 # Target image to build
 #--------------------------------------
-ARG IMAGE=latest
+ARG TARGET=latest
 
 #--------------------------------------
 # Ubuntu base image to use
 #--------------------------------------
-ARG FLAVOR=latest
+ARG FLAVOR=bionic
+ARG BASE_IMAGE=${FLAVOR}
+ARG USER_NAME=ubuntu
+ARG USER_ID=1000
+ARG APP_ROOT=/usr/src/app
 
 #--------------------------------------
 # renovate rebuild trigger
 #--------------------------------------
-FROM renovate/ubuntu@sha256:b20bd58a962952580352f731606c63f621e525cc4706a4b78baf71e852b69da4 as base-latest
-FROM renovate/ubuntu:bionic@sha256:1ab4cb684944bfd76b2a29c368e4e221b796f08fb826c1a826327476cf196647 as base-bionic
-FROM renovate/ubuntu:focal@sha256:b20bd58a962952580352f731606c63f621e525cc4706a4b78baf71e852b69da4 as base-focal
+FROM ubuntu:bionic@sha256:ea188fdc5be9b25ca048f1e882b33f1bc763fb976a8a4fea446b38ed0efcbeba as latest
```

Comment:
I'd like to consider building only `bionic` for now, and having this value as a regular ARG at the top of the file. Will it work?

Reply:
🤔 I need to checkout how to pass the full image (with digest) as arg. Maybe i need some bash / js foo.

* move the trigger images to new file
* on build extract corresponding image from that and pass via BASE_IMAGE arg. 

🤔  can work

Reply:
I am confident that it will be OK to "mix" flavors between sidecar containers and therefore we could simplify all this to support one flavor at a time.

BTW we should also ideally use a generic username too instead of `ubuntu` so that we have more flexibility to switch distributions one day and mix them too. e.g. `/home/user` or `/home/default` or something like that.

Reply:
using user `renovate` and reuse user_id `1000` ?

Reply:
I prefer to keep `renovate` out of it too, as it will likely be used for non-renovate applications too

Reply:
ok, so using `user` 🙃